solr/core/src/test/org/apache/solr/handler/admin/CoreAdminHandlerActionTest.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.handler.admin;
+
+import java.util.Locale;
+import java.util.Map;
+import org.apache.solr.SolrTestCaseJ4;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.params.CoreAdminParams;
+import org.apache.solr.common.util.NamedList;
+import org.apache.solr.common.util.SimpleOrderedMap;
+import org.apache.solr.handler.admin.CoreAdminHandler.CallInfo;
+import org.apache.solr.handler.admin.CoreAdminHandler.CoreAdminOp;
+import org.apache.solr.response.SolrQueryResponse;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class CoreAdminHandlerActionTest extends SolrTestCaseJ4 {
+
+  private static CoreAdminHandler admin = null;
+
+  private static final String SOLRXML =
+      "<solr><coreAdminHandlerActions>"
+          + "<str name='action1'>"
+          + CoreAdminHandlerActionTest.CoreAdminHandlerTestAction1.class.getName()
+          + "</str>"
+          + "<str name='action2'>"
+          + CoreAdminHandlerActionTest.CoreAdminHandlerTestAction2.class.getName()
+          + "</str>"
+          + "</coreAdminHandlerActions></solr>";
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+
+    setupNoCoreTest(createTempDir(), SOLRXML);
```

Comment:
Thank you for the update! I tried exactly the same thing, but my test was failing locally complaining about loader is unable to load the action class :) Magic! I just checked It and it is passed! Arghhh :)